### PR TITLE
clang-tidy: Additions for case checking in source code

### PR DESCRIPTION
**Pull request by sidprice**
This PR adds case checking for various elements of the source code.

These checks were added to assist contributors in checking their submissions against the project contribution requirements.

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -28,6 +28,20 @@ CheckOptions:
     value: '2'
   - key:   readability-identifier-length.MinimumVariableNameLength
     value: '2'
-  - key:   readability-identifier-length.MinimumParameterNameLength
-    value: '2'
```
Comment (dragonmux) [inline .clang-tidy:32]: Did you mean to blow away the minimum parameter name length setting?

**Review by dragonmux — COMMENTED** [review]
Only one item's stood out in review - with that addressed, we're happy to merge this. Thank you for the contribution!

File: .clang-tidy (hunk 1 above)
Comment (sidprice) [reply .clang-tidy:32]: No, copy-pasta error by idiot on keyboard :) Will fix momentarily.

**Review by dragonmux — APPROVED** [review]
LGTM, merging.

**Pull request merged**